client/ayon_core/pipeline/workfile/workfile_template_builder.py

```diff
@@ -835,102 +861,124 @@ def get_template_preset(self):
 
         This can be overridden in host integrations to perform additional
         resolving over the template. Like, `hou.text.expandString` in Houdini.
+        It's recommended to still call the super().resolve_template_path()
+        to ensure the basic resolving is done across all integrations.
 
         Arguments:
             path (str): The input path.
-            fill_data (dict[str, str]): Data to use for template formatting.
+            fill_data (dict[str, str]): Deprecated. This is computed inside
+                the method using the current environment and project settings.
+                Used to be the data to use for template formatting.
 
         Returns:
             str: The resolved path.
 
         """
-        result = StringTemplate.format_template(path, fill_data)
-        if result.solved:
-            path = result.normalized()
+
+        # If the path is an AYON entity URI, then resolve the filepath
+        # through the backend
+        if path.startswith("ayon+entity://") or path.startswith("ayon://"):
+            # This is a special case where the path is an AYON entity URI
+            # We need to resolve it to a filesystem path
+            resolved_path = resolve_entity_uri(path)
+            return resolved_path
+
+        # If the path is set and it's found on disk, return it directly
+        if path and os.path.exists(path):
+            return path
+
+        # Otherwise assume a path with template keys, we do a very mundane
+        # check whether `{` or `<` is present in the path.
+        if "{" in path or "<" in path:
+            # Resolve keys through anatomy
+            project_name = self.project_name
+
+            # Try to fill path with environments and anatomy roots
+            anatomy = Anatomy(project_name)
+            fill_data = {
+                key: value
+                for key, value in os.environ.items()
+            }
+
+            fill_data["root"] = anatomy.roots
+            fill_data["project"] = {
+                "name": project_name,
+                "code": anatomy.project_code,
+            }
+
+            # Recursively remap anatomy paths
+            while True:
+                try:
+                    solved_path = anatomy.path_remapper(path)
+                except KeyError as missing_key:
+                    raise KeyError(
+                        f"Could not solve key '{missing_key}'"
+                        f" in template path '{path}'"
+                    )
+
+                if solved_path is None:
+                    solved_path = path
+                if solved_path == path:
+                    break
+                path = solved_path
+
+            solved_path = os.path.normpath(solved_path)
+            if not os.path.exists(solved_path):
+                return path
+
+            result = StringTemplate.format_template(path, fill_data)
+            if result.solved:
+                path = result.normalized()
+            return path
```

Comment:
@iLLiCiTiT looking at the old code I have a feeling this should be above the `while True:` part... however, I'm not entirely sure why the recursive `anatomy.path_remapper` calls are needed. Do you know? If the string template is filled - why would it still need that recursive resolving?

Am I having a bad morning and am I just not seeing it? 🤯

Reply:
Ah! The fact that it maps full path to rootless makes sense - that als means the format template should indeed be after it.

However, does that mean that the:
```python
        if "{" not in path:
            return path
```
Should also come after the `path_remapper` because it may introduce root keys again, etc?